services/src/main/java/org/keycloak/services/DefaultKeycloakSessionFactory.java

```diff
@@ -229,6 +217,10 @@ private boolean isEnabled(ProviderFactory factory, Config.Scope scope) {
         if (factory instanceof EnvironmentDependentProviderFactory) {
             return ((EnvironmentDependentProviderFactory) factory).isSupported();
         }
+        if (factory instanceof FeatureDependentProviderFactory) {
+            return Optional.ofNullable(((FeatureDependentProviderFactory) factory).getRequiredFeatures()).orElse(Collections.emptySet())
+                    .stream().allMatch(requiredFeature -> Profile.isFeatureEnabled(requiredFeature));
+        }
```

Comment:
@stianst Is it ok to merge this additional feature in common services into this PR, or should it be a separate PR (addition of FeatureDependentProviderFactory, and its handling here)?